src/main/java/seedu/address/model/delivery/DeliveryList.java

```diff
@@ -122,21 +121,46 @@ public ObservableList<Delivery> asUnmodifiableObservableList() {
      * Used when sorting the list by a specified attribute, in descending order.
      */
     public void reverseDeliveryList() {
-        Collections.reverse(internalList);
+        int index = getFirstArchivedIndex().getZeroBased();
+        Collections.reverse(internalList.subList(0, index));
+        Collections.reverse(internalList.subList(index, internalList.size()));
+    }
+
+    /**
+     * Checks if the archive status of two deliveries are the same.
+     */
+    public boolean isSameArchiveStatus(Delivery d1, Delivery d2) {
+        boolean d1Archive = d1.isArchived();
+        boolean d2Archive = d2.isArchived();
+        return (d1Archive && d2Archive) || (!d1Archive && !d2Archive);
```

Comment:
Wait, can this be d1Archive == d2Archive?

Reply:
Good observation!